### PR TITLE
[App] Fix environment check with command redirection

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

- Fixes a bug that can happen if your python command redirects somewhere else
- Fixes the UX if our environment check or version checks fail for some reason so that the command will continue anyway
- Adds tests for the environment check part

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning/app/cli/lightning_cli.py [hunk 1]
```diff
@@ -76,11 +76,18 @@ def main() -> None:
...
+            # Enforce running in PATH Python
+            _check_environment_and_redirect()
+
+            # Check for newer versions and upgrade
+            _check_version_and_upgrade()
+        except SystemExit:
+            raise
+        except Exception:
+            # Note: We intentionally ignore all exceptions here so that we never panic if one of the above calls fails.
+            # If they fail for some reason users should still be able to continue with their command.
```
Comment (awaelchli) [inline src/lightning/app/cli/lightning_cli.py:89]: This also means that we will never know if these checks break in the future. Is this acceptable?

File: tests/tests_app/utilities/test_cli_helpers.py [hunk 2]
```diff
@@ -111,3 +118,51 @@ def test_get_newer_version(mock_requests, releases, current_version, newer_versi
...
+
+
+@patch("lightning.app.utilities.cli_helpers._redirect_command")
+def test_check_environment_and_redirect(mock_redirect_command, tmpdir):
+    original_path = os.environ["PATH"]
+
+    # Ensure that the test fails if it tries to redirect
+    mock_redirect_command.side_effect = RuntimeError
+
+    try:
```
Comment (awaelchli) [inline tests/tests_app/utilities/test_cli_helpers.py:130]: you could use monkeypatch to make the test a bit cleaner (without try-except)

**Review by awaelchli — APPROVED** [review]
@ethanwharris I didn't understand this part: "if your python command redirects somewhere else". For my understanding, can you give an example?

File: tests/tests_app/utilities/test_cli_helpers.py (hunk 2 above)
Comment (ethanwharris) [reply tests/tests_app/utilities/test_cli_helpers.py]: Cool! Yeah that's much better 😃

**Pull request merged**